### PR TITLE
Fix observance of `options.EagerlyEmitFirstEvent`

### LINKED ISSUE #70: EagerlyEmitFirstEvent is ignored
After upgrading to 4.0.0 it appears that EagerlyEmitFirstEvent is ignored.

This unit test fails. Increasing the Task.Delay to 2100 will make this test succeed.

```csharp
    [Fact]
    public async Task EagerlyEmitFirstEventShouldWriteBatchImmediately()
    {
        var eventEmitted = false;
        var bs = new CallbackBatchedSink(_ =>
        {
            eventEmitted = true;
            return Task.CompletedTask;
        });

        var options = new PeriodicBatchingSinkOptions
        {
            Period = TimeSpan.FromSeconds(2),
            EagerlyEmitFirstEvent = true,
            BatchSizeLimit = 10,
            QueueLimit = 1000
        };
        var pbs = new PeriodicBatchingSink(bs, options);
        var evt = Some.InformationEvent();
        pbs.Emit(evt);

        await Task.Delay(1900);
        Assert.True(eventEmitted);

        pbs.Dispose();
    }
```

### PR DESCRIPTION
**Pull request by nblumhardt**
Fixes #70

### CODE REVIEW

**Comment by ArieGato** [comment]
I would also add a test case to test for `EagerlyEmitFirstEvent = false`

something like this (untested):

```csharp
    [Fact]
    public async Task WhenEagerlyEmitFirstEventIsFalseItShouldNotWriteToBatchImmediately()
    {
        var eventEmitted = false;
        var bs = new CallbackBatchedSink(_ =>
        {
            eventEmitted = true;
            return Task.CompletedTask;
        });

        var options = new PeriodicBatchingSinkOptions
        {
            Period = TimeSpan.FromSeconds(2),
            EagerlyEmitFirstEvent = false,
            BatchSizeLimit = 10,
            QueueLimit = 1000
        };

#if FEATURE_ASYNCDISPOSABLE
        await
#endif
        using var pbs = new PeriodicBatchingSink(bs, options);

        var evt = Some.InformationEvent();
        pbs.Emit(evt);

        await Task.Delay(100);
        Assert.False(eventEmitted);
    }
```

File: test/Serilog.Sinks.PeriodicBatching.Tests/PeriodicBatchingSinkTests.cs [hunk 1]
```diff
@@ -109,4 +109,34 @@ public void ExecutionContextDoesNotFlowToBatchedSink()
 
         Assert.Equal(default(int), observed);
     }
+    
+    [Fact]
+    public async Task EagerlyEmitFirstEventShouldWriteBatchImmediately()
```
Comment (ArieGato) [inline test/Serilog.Sinks.PeriodicBatching.Tests/PeriodicBatchingSinkTests.cs]: Maybe change the name to be simular to the other tests

```suggestion
    public async Task WhenEagerlyEmitFirstEventIsTrueItShouldWriteToBatchImmediately()
```

**Comment by nblumhardt** [comment]
Changed to a theory so we can test the true/false cases together 👍

**Review by ArieGato — APPROVED** [review]
Looks good to me

**Review by ArieGato — APPROVED** [review]

**Review by sungam3r — APPROVED** [review]

**Pull request merged**